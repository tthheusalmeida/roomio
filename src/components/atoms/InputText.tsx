"use client";
import { mergeClassNames } from "@/utils/classNames";

interface InputTextProps extends React.InputHTMLAttributes<HTMLInputElement> {
  className?: string;
}

export default function InputText({ className, ...props }: InputTextProps) {
  return (
    <input
      className={mergeClassNames(
        "w-full h-10 rounded border border-slate-300",
        "px-4 py-2",
        "text-base text-gray-600 caret-slate-600",
        "placeholder-gray-400",
        "outline-none transition-all duration-300",
        "focus:ring-1 focus:ring-slate-400 focus:border-slate-400",
        props?.disabled ? "disabled:bg-slate-200 cursor-not-allowed" : "",
        className ?? "",
      )}
      {...props}
    />
  );
}
