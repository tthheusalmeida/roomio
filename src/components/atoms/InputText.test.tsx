import { render, screen } from "@testing-library/react";
import InputText from "./InputText";

describe("InputText component", () => {
  it("renders an input element", () => {
    const { asFragment } = render(<InputText placeholder="Enter name" />);

    const input = screen.getByPlaceholderText("Enter name");
    expect(input).toBeInTheDocument();
    expect(input.tagName).toBe("INPUT");

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies disabled prop and related classes", () => {
    render(<InputText disabled placeholder="Disabled input" />);

    const input = screen.getByPlaceholderText("Disabled input");
    expect(input).toBeDisabled();
    expect(input).toHaveClass("disabled:bg-slate-200", "cursor-not-allowed");
  });

  it("applies extra classes passed via className", () => {
    const { asFragment } = render(
      <InputText className="extra-class" placeholder="Custom class" />
    );

    const input = screen.getByPlaceholderText("Custom class");
    expect(input).toHaveClass("extra-class");

    expect(asFragment()).toMatchSnapshot();
  });
});
