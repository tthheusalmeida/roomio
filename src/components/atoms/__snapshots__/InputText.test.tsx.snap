// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`InputText component applies extra classes passed via className 1`] = `
<DocumentFragment>
  <input
    class="w-full h-10 rounded border border-slate-300 px-4 py-2 text-base text-gray-600 caret-slate-600 placeholder-gray-400 outline-none transition-all duration-300 focus:ring-1 focus:ring-slate-400 focus:border-slate-400 extra-class"
    placeholder="Custom class"
  />
</DocumentFragment>
`;

exports[`InputText component renders an input element 1`] = `
<DocumentFragment>
  <input
    class="w-full h-10 rounded border border-slate-300 px-4 py-2 text-base text-gray-600 caret-slate-600 placeholder-gray-400 outline-none transition-all duration-300 focus:ring-1 focus:ring-slate-400 focus:border-slate-400"
    placeholder="Enter name"
  />
</DocumentFragment>
`;
